### PR TITLE
Udhr update

### PR DESCRIPTION
**Pull request by dougfelt**
Update udhr data, sample generation tools, and samples.

### CODE REVIEW

File: nototools/update_udhr_samples.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def update_udhr(udhr_dir, fetch_dir, in_repo):
 
 def parse_index(src_dir):
   """Parse the index.xml file in src_dir and return a map from bcp to a set of
-  file codes.
+  file codes, and a map from file code to ohchr code.
```
Comment (jungshik) [inline nototools/update_udhr_samples.py]: what's ohchr ?   I can't find that attribute in xml files. I must be mssing something.

Comment (dougfelt) [reply nototools/update_udhr_samples.py]: It's in the index.xml file.

File: nototools/update_udhr_samples.py [hunk 2]
```diff
@@ -113,75 +128,201 @@ def parse_index(src_dir):
...
+      'urd_2': None,
+  },
+
   }
 
 def fix_index(bcp_to_codes):
   """Take a mapping from bcp to a set of file codes, and
   select the mappings we want using a whitelist.  We return
-  a mapping from one bcp47 code to one file code."""
+  a mapping from one bcp47 code to one file code.
```
[the remark this answers is not in the document]
Comment (dougfelt) [reply nototools/update_udhr_samples.py]: Yes.

File: nototools/update_udhr_samples.py [hunk 3]
```diff
@@ -227,44 +368,59 @@ def filter_bcp_to_code(bcp_to_code):
...
-      value = bcp_to_code[OPTION_MAP[lang_scr]]
-      print 'adding %s for %s' % (value, lang_scr)
-      bcp_to_code[lang_scr] = value
-  return bcp_to_code
+      alias = OPTION_MAP[lang_scr]
+      print 'adding %s (from %s)' % (lang_scr, alias)
+      bcp_to_code_attrib_sample[lang_scr] = bcp_to_code_attrib_sample[alias]
+
+  if errors:
+    print 'add default lang script encountered %d errors:' % len(errors)
```
[the remark this answers is not in the document]
Comment (dougfelt) [reply nototools/update_udhr_samples.py]: Yes.

File: nototools/update_udhr_samples.py [hunk 4]
```diff
@@ -280,33 +436,58 @@ def get_code_to_attrib(src_dir):
...
-  bcp_to_code = filter_bcp_to_code(add_likely_scripts(fix_index(bcp_to_code)))
-  bcp_to_code = add_default_lang_script(bcp_to_code)
+def get_bcp_to_code_attrib_sample(src_dir, ohchr_dir):
+  """Return a mapping from bcp to code (for debugging), attribution, and
+  sample.  The process is:
+  1) parse the index.xml file to determine a mapping from bcp to code.
+     the bcp has at least lang and script, and perhaps region/variant.
+     Multiple codes might share the same bcp.
+  2) Use a whitelist to fix cases where a bcp maps to multiple codes,
+     either by selection one code, or assigning a separate bcp value
```
[the remark this answers is not in the document]
Comment (dougfelt) [reply nototools/update_udhr_samples.py]: Thanks for pointing this outl.

**Pull request merged**